ui/app/components/auth-jwt.js

```diff
@@ -0,0 +1,163 @@
+import Ember from 'ember';
+import { inject as service } from '@ember/service';
+import Component from './outer-html';
+import { next, later } from '@ember/runloop';
+import { task, timeout, waitForEvent } from 'ember-concurrency';
+import { computed } from '@ember/object';
+
+const WAIT_TIME = Ember.testing ? 0 : 500;
+const ERROR_WINDOW_CLOSED =
+  'The provider window was closed before authentication was complete.  Please click Sign In to try again.';
+const ERROR_MISSING_PARAMS =
+  'The callback from the provider did not supply all of the required parameters.  Please click Sign In to try again. If the problem persists, you may want to contact your administrator.';
+
+export { ERROR_WINDOW_CLOSED, ERROR_MISSING_PARAMS };
+
+export default Component.extend({
+  store: service(),
+  selectedAuthPath: null,
+  roleName: null,
+  role: null,
+  onRoleName() {},
+  onLoading() {},
+  onError() {},
+  onToken() {},
+  onNamespace() {},
+
+  didReceiveAttrs() {
+    next(() => {
+      let { oldSelectedAuthPath, selectedAuthPath } = this;
+      if (oldSelectedAuthPath !== selectedAuthPath) {
+        this.set('role', null);
+        this.onRoleName(null);
+        this.fetchRole.perform(null, { nodebounce: true });
+      }
+      this.set('oldSelectedAuthPath', selectedAuthPath);
+    });
+  },
+
+  // OIDC roles in the JWT/OIDC backend are those with an authUrl,
+  // those that are JWT type will 400 when trying to fetch the role
+  isOIDC: computed('role', function() {
+    return this.role && this.role.authUrl;
+  }),
```

Comment:
Should probably also add in `role.authUrl` for correctness, but the dependent keys tell ember when to dirty the cache so that the value will be recomputed on the next `get`. In practice, we just want to check the authUrl when the role changes, which is why I left it out, but I'll add it in because `authUrl` is part of the returned value.